module-4/backend/week_two.md

```diff
@@ -5,18 +5,43 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+var vs. let declaration of variables
+
 2. What's the difference between asynchronous and synchronous JavaScript? 
+Async JS deals with certain actions being performed independant of other actions, w/o affecting them and causing page refreshes. It allows you to perform actions in a multi-threaded fashion through the browser vs. sync JS where you perform only one sync action (consisting of series of actions). 
+
 3. What are the four pillars of Object Oriented programming?
+Abstraction, Inheritance, Encapsulation, Polymorphism.
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+Prototypes and es6 syntax for class definition vs. es5 constructor functions.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+Don't repeat yourself, SRP, long functions, long and/or deeply nested conditionals.
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+Callback function is a function that is passed as an arguement to another function. Callbacks can allow you to peform asyc actions.
+
 7. What are the different scopes of variables in Javascript? When would you want to define global variables?
+Global and local variables. You would want to define a global variable if something will not change after it's defined and if it will be utilized more that once.
+
 8. What's the difference between `==` and `===` in JavaScript?
+Equal vs. strict equal. The former checks if the value is the same , whereas the latter checks if the value and object type are the same.
+
 9. Why do front end frameworks exist?
+To have a more appealing way to interact and display content.
```

Comment:
Also does a lot for us behind the scenes so we don't have to keep doing everything in JS from scratch